src/WebJobs.Script.WebHost/Security/KeyManagement/SecretManager.cs

```diff
@@ -30,6 +29,8 @@ public class SecretManager : IDisposable, ISecretManager
         private readonly ISecretsRepository _repository;
         private readonly HostNameProvider _hostNameProvider;
         private readonly StartupContextProvider _startupContextProvider;
+        private readonly Lazy<bool> _strictHISFeatureEnabled = new Lazy<bool>(() => FeatureFlags.IsEnabled(ScriptConstants.FeatureFlagStrictHISModeEnabled));
+        private readonly Lazy<bool> _strictHISWarnFeatureEnabled = new Lazy<bool>(() => FeatureFlags.IsEnabled(ScriptConstants.FeatureFlagStrictHISModeWarn));
```

Comment:
Is the plan that customers set these flags themselves? Are we going to eventually roll "Warn mode" out to everyone via ScmHostingConfig (or just make it the default at some point?)

Reply:
No, we will never be rolling out a change because that would be breaking. Customer need to opt into this, and the two different modes allow them to work towards strict, by first using warn, regenerating the specified keys, then switching to strict. In strict mode, attempts to use non-HIS keys will result in 401s.